### ISSUE TITLE
Integrate a11y-view into the build process

### ISSUE THREAD

**Issue by zepumph** [phetsims/chipper#612]
From a chat with @jessegreenberg on slack let's add the a11y view on build based on #611:

Jesse Greenberg [10:27 AM] 
@mjkauzmann what do you think about adding generate-a11y-view-html to the requirejs build task so it is added to build dir every grunt?

[10:28] 
Would add it to _every_ sim, maybe that is OK? (edited)

Michael Kauzmann [10:29 AM] 
Hmm

Michael Kauzmann [11:30 AM] 
If we wanted to go down that road, would we want to parallel the way that the sim development html is created/maintained? (edited)

Jesse Greenberg [12:13 PM] 
I was thinking it might parallel how sim_en-iframe.html is created  during requirejs build task

[12:15] 
I don't see that one in the list of grunt tasks, not too sure when it is generated

Michael Kauzmann [12:15 PM] 
That sounds reasonable. I think the best way to go would be to create chipper/data/accessibility, so that development a11y views can be updated quickly with `for-each.sh`. In addition we should add it to the build dir on normal build (for all sims? only for accessibility sims?)

[12:16] 
>I don't see that one in the list of grunt tasks, not too sure when it is generated

It isn't a separate task, I think it is part of `after-requirejs-build`

[12:17] 
Maybe if we detect that the sim is an a11y sim (in the list), then we call `generateA11yViewHTML( grunt, buildConfig );` in the after-requirejs-build task and and copy it into the build dir.

Jesse Greenberg [12:20 PM] 
I see, so maybe it would  be a step in createHTMLFiles (run in afterRequirejsBuild)?

Michael Kauzmann [12:20 PM] 
yeah

Jesse Greenberg [12:20 PM] 
That sounds great

Michael Kauzmann [12:21 PM] 
if we want it on build I think that is the best place

Jesse Greenberg [12:21 PM] 
Do you think we need both a list in chipper/data/ and this added to grunt task?

Michael Kauzmann [12:21 PM] 
Do you agree that having both is a good way to proceed

[12:21] 
lol

Jesse Greenberg [12:21 PM] 
hehe

[12:21] 
Does phet-io have  such a list?

[12:22] 
Seems like it might be useful for organization

[12:24] 
I mean for a11y, even if phet-io doesn't have one

Michael Kauzmann [12:24 PM] 
I do for local development. PhET-iO does indeed have one. It is nice as we moved from a managable number of instrumented sims to many (over 10) as I think accessibilty will be doing soon.

Jesse Greenberg [12:25 PM] 
Yes, agreed!

Michael Kauzmann [12:33 PM] 
Great. I can do this stuff if you'd like (I need more accessibility time this week) (edited)

Jesse Greenberg [12:33 PM] 
Thanks @mjkauzmann that would be great!

**Comment by zepumph** [comment]
@jessegreenberg, I updated the build process so that if the sim is in the a11y list, then it will add the a11y view to the build dir. 

There is one foreseeable problem that I have found with my implementation. 

I am running `generateA11yViewHTML` in `createHTMLFiles` (in `after-requirejs-build`). This is generating the top level a11y view in the sim repo, and then copying that into the build directory. In terms of code, this was the easiest solution. Potential Proplem: if the top level a11y view is not up to date with the sim template (which should never happen), then you will end up with a modified file in you working copy from running grunt. This is less than ideal, but not horrible. But weird and confusing since you just did a build command.

2 possible solutions I have thought up:

1. Rework the `generateA11yViewHTML` task so that it can either write to the top level, or just return the html as a string. Then we can just write it where we need it to go.
1. Just commit and push it the a11y view in `createHTMLFiles` if we need to. (not sure if this is the best solution.

We should also note that this edge case really shouldn't happen. We shouldn't ever really have an out of sync top level a11y view html file with the sim template in chipper. 

Can you please review commits and comment.

**Comment by jessegreenberg** [comment]
@zepumph I am really liking option 1 if it is possible.

> it can either write to the top level, or just return the html as a string.

So it would be smart enough to *only* place the file in the build dir if grunt was called, and place it in top level if grunt generate-a11y-view-html was called?

**Comment by zepumph** [comment]
I agree with @pixelzoom it is good for now, and we can work on it more when we are working on new build tools. I would hate to dive in only to have everything change in a month or two.

**Comment by zepumph** [comment]
Another note. It would be nice if the a11y view wasn't copied on phetio build. Since the sim's name is different, it won't work anyways.

**Comment by zepumph** [comment]
Priority: do it in master right now, don't wait for chipper:2.0 work to begin.

**Comment by zepumph** [comment]
I made another file called `getA11yViewHTMLFromTemplate` that reads the template, fills it in, and returns a string. This is used to make the top level html, as well as create on build.  

I also made sure that this doesn't happen on phet-io build.

I think this is ready to close.

**Comment by jessegreenberg** [comment]
Awesome, thanks @zepumph.
